lib/usd/translators/shading/fileTextureUtil.cpp

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "fileTextureUtil.h"
+
+#include "shadingTokens.h"
+
+#include <mayaUsd/fileio/jobs/jobArgs.h>
+
+#include <pxr/base/tf/pathUtils.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/imaging/hio/image.h>
+#include <pxr/usdImaging/usdImaging/tokens.h>
+
+#include <ghc/filesystem.hpp>
+
+#include <regex>
+#include <system_error>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+namespace {
+// Match UDIM pattern, from 1001 to 1999
+const std::regex
+    _udimRegex(".*[^\\d](1(?:[0-9][0-9][1-9]|[1-9][1-9]0|0[1-9]0|[1-9]00))(?:[^\\d].*|$)");
+} // namespace
+
+void FileTextureUtil::MakeUsdTextureFileName(
+    std::string&       fileTextureName,
+    const std::string& usdFileName,
+    bool               isUDIM)
+{
```

Comment:
Refactoring functions common between usdFileTextureWriter and mtlxFileTextureWriter.